### PR TITLE
v3.5: introduce new work scheduler.

### PR DESCRIPTION
**Pull request by gluszczyk**
# Deploy preview

Scroll down to the checks section for the link to the deploy preview of your branch.

# Links

Link your Jira ticket.

- [Jira](https://livechatinc.atlassian.net/browse/API-10879)

# Description

# Review and release

Apply changes and resolve conflicts. Once the described functionality lands on prod, let us know on #platform-docs-releases that your PR is ready for release. We will **merge** and **publish** the changes.

### CODE REVIEW

File: src/pages/management/configuration-api/v3.5/index.mdx [hunk 1]
```diff
@@ -203,13 +203,18 @@ Creates a new agent with specified parameters within a license.
...
-| `work_scheduler.****.<work_scheduler>` | `object`   | No       | `<work_scheduler>` values __\*\*\*\*\*__                                                                                                              |
+| `work_scheduler.timezone`              | `string`   | Yes      | Timezone for the work scheduler. Falls back to the agent's timezone if empty.                                                                         |
+| `work_scheduler.schedule`              | `object[]` | Yes      | List of agent's working hours.                                                                                                                        |
+| `work_scheduler.schedule[].day`        | `string`   | Yes      | The day for which the working hours are set. Possible values: sunday, monday, tuesday, wednesday, thursday, friday, and saturday.                     |
+| `work_scheduler.schedule[].enabled`    | `bool`     | Yes      | Whether the given set of working hours is enabled                                                                                                     |
```
Comment (kamilatokarska) [inline src/pages/management/configuration-api/v3.5/index.mdx:213]:
```suggestion
| `work_scheduler.schedule[].enabled`    | `bool`     | Yes      | Whether the given set of working hours is enabled.                                                                                                     |
```

File: src/pages/management/configuration-api/v3.5/index.mdx [hunk 2]
```diff
@@ -203,13 +203,18 @@ Creates a new agent with specified parameters within a license.
...
+| `work_scheduler.schedule[].start`      | `string`   | Yes      | When working hours start                                                                                                                              |
```
Comment (kamilatokarska) [inline src/pages/management/configuration-api/v3.5/index.mdx:214]:
```suggestion
| `work_scheduler.schedule[].start`      | `string`   | Yes      | When working hours start.                                                                                                                              |
```

File: src/pages/management/configuration-api/v3.5/index.mdx [hunk 3]
```diff
@@ -203,13 +203,18 @@ Creates a new agent with specified parameters within a license.
...
+| `work_scheduler.schedule[].end`        | `string`   | Yes      | When working hours end                                                                                                                                |
```
Comment (kamilatokarska) [inline src/pages/management/configuration-api/v3.5/index.mdx:215]:
```suggestion
| `work_scheduler.schedule[].end`        | `string`   | Yes      | When working hours end.                                                                                                                                |
```

File: src/pages/management/changelog/index.mdx [hunk 4]
```diff
@@ -33,6 +33,7 @@ The developer preview version provides a preview of the upcoming changes to the
 
 - The [**List Agents**](/management/configuration-api/v3.5/#list-agents) method can now filter agents by their `suspended` status.
 - The following methods now support batch requests: [Create Agent](/management/configuration-api/v3.5/#create-agent), [Update Agent](/management/configuration-api/v3.5/#update-agent), [Delete Agent](/management/configuration-api/v3.5/#delete-agent), [Suspend Agent](/management/configuration-api/v3.5/#suspend-agent), [Unsuspend Agent](/management/configuration-api/v3.5/#unsuspend-agent), and [Approve Agent](/management/configuration-api/v3.5/#approve-agent).
+- `work_scheduler` parameter in [**Create Agent**](/management/configuration-api/v3.5/#create-agent), [**Update Agent**](/management/configuration-api/v3.5/#update-agent), [**Create Bot**](/management/configuration-api/v3.5/#create-bot), and [**Update Bot**](/management/configuration-api/v3.5/#update-bot) methods has a new format. It is now possible to configure multiple working hours for each day.
```
Comment (kamilatokarska) [inline src/pages/management/changelog/index.mdx:36]:
```suggestion
- The `work_scheduler` parameter in [**Create Agent**](/management/configuration-api/v3.5/#create-agent), [**Update Agent**](/management/configuration-api/v3.5/#update-agent), [**Create Bot**](/management/configuration-api/v3.5/#create-bot), and [**Update Bot**](/management/configuration-api/v3.5/#update-bot) methods has a new format. It is now possible to configure multiple working hours for each day.
```

File: src/pages/management/changelog/index.mdx [hunk 5]
```diff
@@ -49,6 +50,7 @@ The developer preview version provides a preview of the upcoming changes to the
   - [**tag_created**](/management/webhooks/v3.5/#tag_created)
   - [**tag_deleted**](/management/webhooks/v3.5/#tag_deleted)
   - [**tag_updated**](/management/webhooks/v3.5/#tag_updated)
+- `work_scheduler` field in [**agent_created**](/management/webhooks/v3.5/#agent_created), [**agent_updated**](/management/webhooks/v3.5/#agent_updated), [**bot_created**](/management/webhooks/v3.5/#bot_created), and [**bot_updated**](/management/webhooks/v3.5/#bot_updated) webhooks has a new format.
```
Comment (kamilatokarska) [inline src/pages/management/changelog/index.mdx:53]:
```suggestion
- The `work_scheduler` field in [**agent_created**](/management/webhooks/v3.5/#agent_created), [**agent_updated**](/management/webhooks/v3.5/#agent_updated), [**bot_created**](/management/webhooks/v3.5/#bot_created), and [**bot_updated**](/management/webhooks/v3.5/#bot_updated) webhooks has a new format.
```

File: src/pages/messaging/agent-chat-api/changelog/index.mdx [hunk 6]
```diff
@@ -36,6 +36,7 @@ The developer preview version provides a preview of the upcoming changes to the
   - [**tag_created**](/messaging/agent-chat-api/v3.5/rtm-pushes/#tag_created)
   - [**tag_deleted**](/messaging/agent-chat-api/v3.5/rtm-pushes/#tag_deleted)
   - [**tag_updated**](/messaging/agent-chat-api/v3.5/rtm-pushes/#tag_updated)
+- `work_scheduler` field in [**agent_created**](/messaging/agent-chat-api/v3.5/rtm-pushes/#agent_created), [**agent_updated**](/messaging/agent-chat-api/v3.5/rtm-pushes/#agent_updated), [**bot_created**](/messaging/agent-chat-api/v3.5/rtm-pushes/#bot_created), and [**bot_updated**](/messaging/agent-chat-api/v3.5/rtm-pushes/#bot_updated) pushes has a new format.
```
Comment (kamilatokarska) [inline src/pages/messaging/agent-chat-api/changelog/index.mdx:39]:
```suggestion
- The `work_scheduler` field in [**agent_created**](/messaging/agent-chat-api/v3.5/rtm-pushes/#agent_created), [**agent_updated**](/messaging/agent-chat-api/v3.5/rtm-pushes/#agent_updated), [**bot_created**](/messaging/agent-chat-api/v3.5/rtm-pushes/#bot_created), and [**bot_updated**](/messaging/agent-chat-api/v3.5/rtm-pushes/#bot_updated) pushes has a new format.
```

**Review by litwinow — APPROVED** [review]

**Review by kamilatokarska — APPROVED** [review]
:)

**Review by ale-kacperczyk — APPROVED** [review]

**Comment by litwinow** [comment]
@ale-kacperczyk @kamilatokarska this can now be deployed

**Pull request merged**